frontend/src/metabase/admin/settings/components/SettingsLdapForm.tsx

```diff
@@ -31,118 +37,53 @@ const testParentheses: TestConfig<string | null | undefined> = {
     (value?.match(/\(/g) || []).length === (value?.match(/\)/g) || []).length,
 };
 
-const testPort: TestConfig<string | null | undefined> = {
-  name: "test-port",
-  message: "That's not a valid port number",
-  test: (value) => Boolean((value || "").trim().match(/^\d*$/)),
-};
-
 const LDAP_SCHEMA = Yup.object({
-  "ldap-port": Yup.string().nullable().test(testPort),
+  "ldap-port": Yup.number().nullable(),
   "ldap-user-filter": Yup.string().nullable().test(testParentheses),
   "ldap-group-membership-filter": Yup.string().nullable().test(testParentheses),
 });
 
-export type SettingValues = Partial<Settings>;
-
-type LdapFormValues = Omit<SettingValues, "ldap-port"> & {
-  "ldap-port"?: string;
-};
-
-type LdapFormSettingElement = Omit<SettingElement, "key"> & {
-  key: string; // ensuring key is required
-  is_env_setting?: boolean;
-  env_name?: string;
-  default?: any;
-};
-
-type Props = {
-  elements: LdapFormSettingElement[];
-  settingValues: SettingValues;
-  onSubmit: (values: SettingValues) => void;
-};
-
-export const SettingsLdapFormView = ({
-  elements = [],
-  settingValues,
-  onSubmit,
-}: Props) => {
-  const isEnabled = settingValues["ldap-enabled"];
-
-  const settings = useMemo(() => {
-    return _.indexBy(elements, "key");
-  }, [elements]);
-
-  const fields = useMemo(() => {
-    return _.mapObject(settings, (setting) => ({
-      name: setting.key,
-      label: setting.display_name,
-      description: setting.description,
-      placeholder: setting.is_env_setting
-        ? t`Using ${setting.env_name}`
-        : setting.placeholder || setting.default,
-      default: setting.default,
-      required: setting.required,
-      autoFocus: setting.autoFocus,
-    }));
-  }, [settings]);
-
-  const defaultableAttrs = useMemo(() => {
-    return new Set(PLUGIN_LDAP_FORM_FIELDS.defaultableFormFieldAttributes);
-  }, []);
-
-  const ldapAttributes = useMemo(
-    () => [
-      ...PLUGIN_LDAP_FORM_FIELDS.formFieldAttributes,
-
-      // Server Settings
-      "ldap-host",
-      "ldap-port",
-      "ldap-security",
-      "ldap-bind-dn",
-      "ldap-password",
-
-      // User Schema
-      "ldap-user-base",
-      "ldap-user-filter",
-
-      // Attributes
-      "ldap-attribute-email",
-      "ldap-attribute-firstname",
-      "ldap-attribute-lastname",
-
-      // Group Schema
-      "ldap-group-sync",
-      "ldap-group-base",
-      "ldap-group-membership-filter",
-      "ldap-sync-admin-group",
-    ],
-    [],
-  ) as SettingKey[];
-
-  const attributeValues = useMemo(() => {
-    return getAttributeValues(
-      ldapAttributes,
-      settings,
-      settingValues,
-      defaultableAttrs,
-    );
-  }, [settings, settingValues, ldapAttributes, defaultableAttrs]);
```

Comment:
this is all just an insane amount of code and indirection for the sake of defining these form fields in data instead of in the form below